tablecodec/tablecodec.go

```diff
@@ -469,6 +469,28 @@ func appendTableIndexPrefix(buf []byte, tableID int64) []byte {
 	return buf
 }
 
+// IsRecordKey check whether key is a recordKey.
+func IsRecordKey(key []byte) bool {
```

Comment:
We need to replace tableID for index key as well.
Just use `bytes.HasPrefix(key, tablePrefix)` will do.